nevergrad/functions/perfcap3d/core.py

```diff
@@ -0,0 +1,436 @@
+            # ["DOFs"] is a boolean list of length 3, indicating which one of the joint axis
+            # are considered for rotation. We can disable degrees of freedom per joint by
+            # inserting False to the related position index inside this list
+            # mutation_variances/bounds_min/bounds_max/mutable_mutation_variances
+            # are arrays of lenth equal to the number of DOFs of the joint (i.e number of True values in ["DOFs"] list)
+            assert len(v["DOFs"]) == 3
+            nvars = sum(map(lambda x: 1 if x else 0, v["DOFs"]))
+
+            assert len(v["mutation_variances"]) == nvars
+            assert len(v["bounds_min"]) == nvars
+            assert len(v["bounds_max"]) == nvars
+            assert len(v["mutable_mutation_variances"]) == nvars
+
+        initial_pose = self._initial_pose
+        dofs_flattened = list(itertools.chain(*map(lambda x: x["DOFs"], variables)))
+        jids_flattened = list(itertools.chain(*map(lambda x: [x["joint_id"]]*len(list(filter(None, x["DOFs"]))), variables)))
+
+        var_indices_flattened = list(range(len(jids_flattened)))
+        dof_indices_flattened = list(itertools.chain(*map(lambda x: [index for index, v in enumerate(x["DOFs"]) if v], variables)))
+        self._variable_to_dof_index_map = dict(zip(var_indices_flattened, dof_indices_flattened))
+        self._variable_to_joint_index_map = dict(zip(var_indices_flattened, jids_flattened))
+        initial_values = list(itertools.compress(itertools.chain(*(initial_pose[jid] for jid in jids)), dofs_flattened))
+
+        # initial_values
+
+        bounds_min = list(itertools.chain(*map(lambda x: x["bounds_min"], variables)))
+        bounds_max = list(itertools.chain(*map(lambda x: x["bounds_max"], variables)))
+        mutation_variances = list(itertools.chain(*map(lambda x: x["mutation_variances"], variables)))
+        mutable_mut_var = list(itertools.chain(*map(lambda x: x["mutable_mutation_variances"], variables)))
+        # parameter normalizers, one for each parameter
+        self._var_normalizers = [VarNormalizer(lower_bound=x[0], upper_bound=x[1], normalize=self._normalize)
+                                 for x in zip(bounds_min, bounds_max)]
+        # parameter creation (initial value, bounds, mutable_mutation_var)
+        params = [p.Scalar(init=n.normalize(init), lower=n.active_lower_bound, upper=n.active_upper_bound, mutable_sigma=ms)
+             for (n, ms, init) in zip(self._var_normalizers, mutable_mut_var, initial_values)]
+        # leave negative mutation variances to auto / ie. do not set
+        for (v, mv, n) in filter(lambda x: x[1] > 0.0, zip(params, mutation_variances, self._var_normalizers)):
+            v.set_mutation(n.normalize(mv), custom="gaussian")      # set variable mutation
+
+        return params
+
+    def __init__(self, server_comm: 'Perfcap3DServerComm', objective_func_cfg: tp.Dict[str, tp.Any], experiment_tag_id: str):
```

Comment:
it does look to me like this could take a mere index of the experiment to use, and then everything could be set up automatically. Can you detail why not?

Reply:
I followed a similar way. Initially it takes a filename and follows configuration from there.